views/components/Matrix.d3.js

```diff
@@ -426,15 +425,17 @@ export default class Matrix {
 
     this.xScaleLinearTop = d3
       .scaleLinear()
-      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1]])
+      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1 || 0]])
       .domain([0, xAxisForDatesData.length - 1])
     this.xScaleLinearBottom = d3
       .scaleLinear()
-      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1]])
-      .domain([startDay, xAxisForDatesData[xAxisForDatesData.length - 1].day])
+      .range([cellWidth / 2, xAxisRange[xAxisRange.length - 1 || 0]])
+      .domain(
+        [startDay, xAxisForDatesData[xAxisForDatesData.length - 1]?.day] || 0
```

Comment:
should the `|| 0` go inside the array?

```suggestion
        [startDay, xAxisForDatesData[xAxisForDatesData.length - 1]?.day || 0]
```

Reply:
When there is no data xAxisForDatesData is empty, if we do this `xAxisForDatesData[0]` that will result in undefined, and we need a value.

Reply:
yeah, but don't we want the value in the array?

Reply:
we want `[startDay, 0]` not `[startDay, undefined]` right?

Reply:
oof, good catch it's still wrapped in the array one
👍

Reply:
[64f55cf](https://github.com/AMP-SCZ/dpdash/pull/634/commits/64f55cf65b019cb2b9f82c6f797eb8449067a8ad)
👍